src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```
@@ -1256,6 +1256,20 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ProjectCapability Include="CrossPlatformExecutable" />
   </ItemGroup>
 
+  <!-- Native AOT compilation -->
+  <ItemGroup>
+    <ProjectCapability Include="NativeAOTProperties" Condition="'$(ShowNativeAOTProperties)' == 'true'"/>
+  </ItemGroup>
+
+  <!-- Native AOT properties should be shown by default for projects targeting .NET 8 or higher, except for WPF and WinForms projects -->
+  <PropertyGroup>
+    <ShowNativeAOTProperties Condition="'$(UseWPF)' != 'true'
+                             and '$(UseWindowsForms)' != 'true'
+                             and '$(ShowNativeAOTProperties)' == ''
+                             and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'
+                             and '$(_TargetFrameworkVersionWithoutV)' >= '8.0'">true</ShowNativeAOTProperties>
```

```suggestion
                             and $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net8.0'))">true</ShowNativeAOTProperties>
```

Is an easier way to make this check.

---

what are user-defined framework aliases?

---

ok, if that's the case, then we will need to fix a few places across the stack that are already using this check.

https://github.com/dotnet/sdk/blob/c9330fb79615cfb6b6b9ab399a496944be62ea95/src/WebSdk/Publish/Targets/Microsoft.NET.Sdk.Publish.targets#L97-L100

https://github.com/dotnet/sdk/blob/c9330fb79615cfb6b6b9ab399a496944be62ea95/src/Containers/packaging/build/Microsoft.NET.Build.Containers.targets#L15

https://github.com/dotnet/runtime/blob/3c3f49958f93edba4e899f292a876bd47fdbeca6/src/libraries/Microsoft.Extensions.Configuration.Binder/src/buildTransitive/Microsoft.Extensions.Configuration.Binder.targets#L14-L21

---

cc @captainsafia @vlada-shubina @baronfel @rainersigwald @layomia

---

In addition to what was already mentioned by @KalleOlaviNiemitalo, evaluation performance is slightly better with the property comparison instead of calling into the `IsTargetFrameworkCompatible` msbuild function.